jvm/src/test/scala/scala/xml/XMLTest.scala

```diff
@@ -749,4 +749,20 @@ class XMLTestJVM {
     assertEquals("<node>\n  <leaf/>\n</node>", pp.format(x))
   }
 
+  @UnitTest
+  def issue231: Unit = {
+    val pp = new xml.PrettyPrinter(4, 2, minimizeEmpty = true)
+    val x = <a b="c"/>
+    val formatted = pp.format(x)
+    assertEquals(x, XML.loadString(formatted))
```

Comment:
Shouldn't this test be comparing strings?  With `XML.loadString`, I think you're just verifying the XML structure is preserved from the original.

Reply:
I'm verifying that there are no extra nodes caused by the formatting, such as the `Text` node reported in #231.

I didn't want to hard-code a string to avoid having to modify the test if the formatting behaviour changes. For example, if the commented-out `todo` is fixed, the output may have 3 lines instead of 2.
```scala
// current behaviour
pp.format(x) == s"""<a\n  b="c"/>"""
// possible future behaviour
pp.format(x) == s"""<a\n  b="c"\n  />"""
```
Since the bug is about the XML structure changing (and, in my case, failing schema validation), I think that using `XML.loadString` is a better option.

What do you think?

Reply:
Ah, ok.  That's a fair point.  The XML structure was getting changed.

There are other tests of `PrettyPrinter` that compare strings.  If history is any guide, I don't think it's a huge risk for the formatting behavior to change often.  Again, strings are the output of `PrettyPrint.format`, so it seems worth checking.

Reply:
Ah, ok.  That's a fair point.  The XML structure was getting changed.

There are other tests of `PrettyPrinter` that compare strings.  If history is any guide, I don't think it's a huge risk for the formatting behavior to change often.  Again, strings are the output of `PrettyPrint.format`, so it seems worth checking.

Reply:
I'm not sure what value a string comparison would add. I think that the functionality of formatting is covered by other tests, e.g. `issue90` on line 746. This test also checks that some formatting happened by checking the number of lines in the output.

I'll add the string comparison if you deem it necessary, but I don't think it would add much value.